packages/dynamic/src/assignInlineVars.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  walkObject,
+  get,
+  getVarName,
+  Contract,
+  MapLeafNodes,
+} from '@vanilla-extract/private';
+
+type Styles = { [cssVarName: string]: string };
+
+export function assignInlineVars(vars: Record<string, string>): Styles;
+export function assignInlineVars<ThemeContract extends Contract>(
+  contract: ThemeContract,
+  tokens: MapLeafNodes<ThemeContract, string>,
+): Styles;
+export function assignInlineVars(varsOrContract: any, tokens?: any) {
+  const styles: { [cssVarName: string]: string } = {};
+
+  if (typeof tokens === 'object') {
+    const contract = varsOrContract;
+
+    walkObject(tokens, (value, path) => {
+      const varName = get(contract, path);
+
+      styles[varName.substring(4, varName.length - 1)] = String(value);
```

Comment:
Is the `substr` logic the same as what `getVarName` does? If so, maybe replace it with that.

Reply:
The original idea here was that it's faster since we already know it's a vanilla-extract contract, but in hindsight this feels like an over-optimisation at the expense of consistent variable handling. If we update the `getVarName` logic in the future (e.g. adding support for vars with fallbacks as keys), this would need to keep up.